features/netsocket/TCPSocket.cpp

```diff
@@ -188,14 +188,8 @@ nsapi_size_or_error_t TCPSocket::recv(void *data, nsapi_size_t size)
 
 void TCPSocket::event()
 {
-    int32_t wcount = _write_sem.wait(0);
-    if (wcount <= 1) {
-        _write_sem.release();
-    }
-    int32_t rcount = _read_sem.wait(0);
-    if (rcount <= 1) {
-        _read_sem.release();
-    }
+    _write_sem.release();
+    _read_sem.release();
```

Comment:
@geky did the events need to be coalesced here to prevent a bug with the esp8266?